tests/integration/telemetry/stats/prometheus/TCP/stats_tcp_filter_test.go

```diff
@@ -0,0 +1,194 @@
+			)
+
+			systemNM := namespace.ClaimSystemNamespaceOrFail(ctx, ctx)
+			mxc, err := file.AsString(metadataExchangeFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", metadataExchangeFilterConfig, err)
+			}
+			stats, err := file.AsString(statsFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", statsFilterConfig, err)
+			}
+
+			cleanup, err := file.AsString(cleanupFilterConfig)
+			if err != nil {
+				t.Errorf("unable to load config %s, err:%v", metadataExchangeFilterConfig, err)
+			}
+
+			g.ApplyConfigOrFail(
+				t,
+				systemNM,
+				mxc,
+				stats,
+				cleanup,
+			)
+			defer g.DeleteConfig(
+				systemNM,
+				mxc,
+				stats,
+				cleanup,
+			)
+
+			util.AllowRuleSync(t)
+
+			destinationQuery := buildQuery()
+			retry.UntilSuccessOrFail(t, func() error {
+				util.SendTraffic(ing, t, "Sending traffic", url, "", 200)
+				// TODO(gargnupur): Use TCP metrics like in Telemetry V1 (https://github.com/istio/istio/issues/20283)
+				if err := util_prometheus.QueryPrometheus(t, destinationQuery, prom); err != nil {
+					t.Logf("prometheus values for istio_requests_total: \n%s", util.PromDump(prom, "istio_requests_total"))
+					return err
+				}
+				return nil
+			}, retry.Delay(3*time.Second), retry.Timeout(80*time.Second))
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("stats_tcp_filter", m).
+		RequireEnvironment(environment.Kube).
+		Label(label.CustomSetup).
+		SetupOnEnv(environment.Kube, istio.Setup(&ist, nil)).
+		Setup(testsetup).
+		Run()
+}
+
+func testsetup(ctx resource.Context) (err error) {
+	bookinfoNs, err = namespace.New(ctx, namespace.Config{
+		Prefix: "istio-bookinfo",
+		Inject: true,
+	})
+	if err != nil {
+		return
+	}
+	if _, err := bookinfo.Deploy(ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookInfo}); err != nil {
+		return err
+	}
+	if _, err = bookinfo.Deploy(ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookinfoRatingsv2}); err != nil {
+		return err
+	}
+	if _, err = bookinfo.Deploy(ctx, bookinfo.Config{Namespace: bookinfoNs, Cfg: bookinfo.BookinfoDb}); err != nil {
+		return err
+	}
+	g, err = galley.New(ctx, galley.Config{})
+	if err != nil {
+		return err
+	}
+	if _, err = mixer.New(ctx, mixer.Config{Galley: g}); err != nil {
+		return err
+	}
+	ing, err = ingress.New(ctx, ingress.Config{Istio: ist})
+	if err != nil {
+		return err
+	}
+	prom, err = prometheus.New(ctx)
+	if err != nil {
+		return err
+	}
+	yamlText, err := bookinfo.NetworkingBookinfoGateway.LoadGatewayFileWithNamespace(bookinfoNs.Name())
+	if err != nil {
+		return err
+	}
+	err = g.ApplyConfig(bookinfoNs, yamlText)
+	if err != nil {
+		return err
+	}
+
+	meshConfig := mesh.DefaultMeshConfig()
+	meshConfig.MixerReportServer = ""
+
+	return nil
+}
+
+func buildQuery() (destinationQuery string) {
+	destinationQuery = `istio_requests_total{reporter="destination",`
```

Comment:
Why not verify source side telemetry?

Reply:
Because we don't get client complete destination info in client side telemetry yet because of https://github.com/envoyproxy/envoy-wasm/issues/291. Will add once this bug is fixed